lang/lowering/src/symbol_table/mod.rs

```diff
@@ -0,0 +1,38 @@
+use ast::HashMap;
+use decls::*;
+use ident::Ident;
+use parser::cst::*;
+use url::Url;
+
+pub mod build;
+pub mod lookup;
+
+/// The symbol table for a single module.
+pub type ModuleSymbolTable = HashMap<Ident, DeclMeta>;
+
+/// The symbol table for a module and all of its imported modules.
+#[derive(Debug, Default, Clone)]
+pub struct SymbolTable {
+    // Maps modules to their respective symbol tables.
+    map: HashMap<Url, ModuleSymbolTable>,
```

Comment:
Might want to use `Arc<ModuleSymbolTable>` here.

Reply:
Will do in the next PR :)